servers/physics_2d/godot_shape_2d.cpp

```diff
@@ -370,7 +370,7 @@ void GodotCapsuleShape2D::get_supports(const Vector2 &p_normal, Vector2 *r_suppo
 
 	real_t d = n.y;
 
-	if (Math::abs(d) < (1.0 - _SEGMENT_IS_VALID_SUPPORT_THRESHOLD)) {
+	if (height > 2.0 * radius && Math::abs(d) < (1.0 - _SEGMENT_IS_VALID_SUPPORT_THRESHOLD)) {
```

Comment:
Might be worth adding some comments, what you explained in the PR/issue isn't immediately obvious when reading this.

Also, what happens if `height < 2.0 * radius`?  Is it valid to handle that as a sphere?

Reply:
In a capsule there are balls of radius `radius` centered at the centers of end caps of a cylinder with radius `radius`, and the `height` is counted including the balls, so `height >= 2.0 * radius` in general; the only degenerate case is `height == 2.0 * radius`. In Godot the `>=` seems to be enforced (only?) by the inspector: trying to make `height` smaller than `2.0 * radius` will decrease the radius, etc.

I think the code is pretty understandable once you see that there are two cases for the `if` statement and the first case depends on the cylinder (respectively rectangle) part of the capsule having a nonzero (half-)height, so I'll just make `h = height * 0.5 - radius` a local variable to be used in both branches, and add a comment to its definition.